src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2155542587>#2011d9</a><span> - 07/06/2024, 09:06 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The method setCapacity(int newCapacity) lacks logic to handle the reduction of capacity which leads to potential memory leaks as it does not remove excess entries.
</div>
<div id="fix">
<b>Fix:</b> Implement logic within setCapacity to remove entries from the cache when the new capacity is less than the current size of the cache.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -22,5 +22,10 @@
      private void setCapacity(int newCapacity) {
          checkCapacity(newCapacity);
 +        while (data.size() > newCapacity) {
 +            K leastUsedKey = getLeastUsedKey();
 +            data.remove(leastUsedKey);
 +        }
          this.cap = newCapacity;
      }
</code></pre>
</div>
</details>
</div>
<br>


</div>